aten/src/ATen/native/NamedTensor.cpp

```diff
@@ -16,5 +16,141 @@ Tensor set_names(const Tensor& self, optional<DimnameList> names) {
   return result;
 }
 
+static void report_moving_unnamed_dim_error(
+    DimnameList names, DimnameList other, bool is_aligning_two_tensors) {
+  if (is_aligning_two_tensors) {
+    TORCH_CHECK(false,
+        "Aligning Tensor", names, " and Tensor", other,
+        " would change the absolute position from the right of an unnamed dimension. ",
+        "Please name unnamed dimensions to avoid ambiguity.");
+  } else {
+    TORCH_CHECK(false,
+        "Aligning Tensor", names, " to `names` ", other,
+        " would change the absolute position from the right of an unnamed dimension. ",
+        "Please name unnamed dimensions to avoid ambiguity.");
+  }
+}
+
+static void report_not_a_subsequence_error(
+    DimnameList names, DimnameList other, bool is_aligning_two_tensors) {
+  if (is_aligning_two_tensors) {
+    auto shorter = names.size() > other.size() ? other : names;
+    auto longer = names.size() > other.size() ? names : other;
+    TORCH_CHECK(false,
+        "Could not align Tensor", shorter, " and Tensor", longer,
```

Comment:
same here ... or does the string representation of `shorter` start with a space?